eoscompanion/main.py

```diff
@@ -149,17 +149,79 @@ def do_dbus_register(self, connection, object_path):  # pylint: disable=argument
 
     def do_dbus_unregister(self, connection, object_path):  # pylint: disable=arguments-differ
         '''Invoked when we lose a D-Bus connection.'''
-        log('Lost session d-bus connection at {path}'.format(path=object_path))
+        logging.warning('Lost session d-bus connection at %s', object_path)
         return Gio.Application.do_dbus_unregister(self,
                                                   connection,
                                                   object_path)
 
     def do_activate(self):  # pylint: disable=arguments-differ
         '''Invoked when the application is activated.'''
-        log('Activated')
+        logging.info('Activated')
         return Gio.Application.do_activate(self)
 
 
+COMPANION_APP_CONFIG_FILES = [
+    '/var/run/host/etc/eos-companion-app/config.ini',
+    '/var/run/host/usr/share/eos-companion-app/config.ini',
+    # FIXME: /var currently does not bind in /run/host in the sandbox
+    '/var/run/host/var/lib/eos-companion-app/config.ini'
+]
+COMPANION_APP_CONFIG_SECTION = 'Companion App'
+LOGLEVEL_CONFIG_NAME = 'loglevel'
+DEFAULT_LOG_LEVEL = logging.INFO
+
+
+def find_best_matching_config_file():
+    '''Get a GKeyFile structure for the first config file.
+
+    This searches all three config directories in priority order, similar
+    to the way that eos-companion-app-configuration-manager does it.
+    '''
+    keyfile = GLib.KeyFile()
+
+    for config_file_candidate_path in COMPANION_APP_CONFIG_FILES:
+        try:
+            keyfile.load_from_file(config_file_candidate_path,
+                                   GLib.KeyFileFlags.NONE)
+        except GLib.Error as error:
+            if error.matches(GLib.file_error_quark(), GLib.FileError.NOENT):
+                continue
+
+            raise error
+
+        return keyfile
+
+    return None
+
+
+def get_log_level():
+    '''Get the log level
+
+    We try to get the log level from the eos-companion-app
+    config file. If we can not get the value we return the
+    default log level.
+    '''
+    keyfile = find_best_matching_config_file()
+
+    if not keyfile:
+        return DEFAULT_LOG_LEVEL
+
+    try:
+        config_level = keyfile.get_value(COMPANION_APP_CONFIG_SECTION,
```

Comment:
Should we not use one of the "typed" getters such as `keyfile.get_integer()` or `keyfiile.get_string()` here instead? It's not actually clear to me what the configuration file format is. The default configuration file that gets installed in `/usr` should also be changed in this patchset to include the new default.

Reply:
`keyfile.get_string()` is what we want here.

I think it is okay to leave it there without the option - the falling back to the hardcoded default is fine.